PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/util/AzureFunctionsUtils.java

```diff
@@ -83,15 +98,26 @@ public static File createMavenProjectToTempFolder(final String groupId, final St
             maps.put("-Dversion", version);
             maps.put("-Dpackage", packageName);
             final String args = maps.entrySet().stream()
-                .map(e -> e.getKey() + "=" + e.getValue())
-                .collect(Collectors.joining(" "));
+                    .map(e -> e.getKey() + "=" + e.getValue())
+                    .collect(Collectors.joining(" "));
             CommandUtils.executeMultipleLineOutput(mvnCmd.getAbsolutePath() + " archetype:generate" + " -B " + args, folder);
 
             return folder;
         }
         return null;
     }
 
+    public static String generateFunctionClassByTrigger(String trigger, String packageName, String className) throws IOException {
```

Comment:
Shall we keep a constant map including the default parameters instead of template files? Or we will have two copies of function template, in IntelliJ and in tools-common, besides, shall we move the default constants to `azure-tools-common` ;ater?

Reply:
we need to abandon mvn archetype since it is very slow on fresh machine